snuba/query/processors/prewhere.py

```diff
@@ -46,13 +51,29 @@ def __init__(
         self.__prewhere_candidates = prewhere_candidates
         self.__omit_if_final = omit_if_final
         self.__max_prewhere_conditions: Optional[int] = max_prewhere_conditions
+        self.__uniq_matcher = FunctionPattern(String("uniq"), None)
 
     def process_query(self, query: Query, request_settings: RequestSettings) -> None:
         max_prewhere_conditions: int = (
             self.__max_prewhere_conditions or settings.MAX_PREWHERE_CONDITIONS
         )
         prewhere_keys = self.__prewhere_candidates
 
+        uniq_cols: Set[str] = set()
+        expressions = query.get_all_expressions()
+        for exp in expressions:
+            match = self.__uniq_matcher.match(exp)
+            if match is not None:
+                columns = get_columns_in_expression(exp)
+                for c in columns:
+                    uniq_cols.add(c.column_name)
+
+        for col in uniq_cols:
+            if col in prewhere_keys:
+                metrics.increment("uniq_col_in_prewhere_candidate")
```

Comment:
I suspect this will be logged a lot because `prewhere_keys` are jsut the candidate columns, not necessarily the columns that will be chosen.

Reply:
Would be good to tag the col so we can decide what to do about those queries

Reply:
ok